### DISCUSSION TITLE
Printer support on Commodore machines

### DISCUSSION THREAD

**Discussion by kcasteels** [cc65/cc65#2531] [Q&A]
I have a BBS program written using CC65 for the C64 and would like to print to an IEC printer (device 4). Does CC65 support printers? In BASIC this is as simple as opening device 4 and printing to that device.

Thanks for your help.

**Comment by mrdudz** [comment]
no specific printer support. for CBM devices, i'd do exactly what you say, using the provided kernal wrapper functions

**Comment by kcasteels** [comment]
So how would I go about calling these kernal functions from C?

**Comment by mrdudz** [comment]
check the cbm.h header, it has all those wrapper functions, they work exactly like the respective kernal functions

**Reply by Marrin** [reply]
You don't even have to go to the kernel level functions, there are BASIC like functions too.  Untested:

```c
char *text = "hello, world!\n";
cbm_open(1, 4, 0, "");  /* OPEN1,4,0,"" */
cbm_write(1, text, strlen(text));  /* PRINT#1,"hello, world" */
cbm_close(1);  /* CLOSE1 */
```

**Comment by polluks** [comment]
We have full ANSI and POSIX support: nothing ;-)
